collector/kernel/kernel_collector_test.cc

```diff
@@ -210,70 +240,117 @@ class KernelCollectorTest : public ::testing::Test {
     stop_test_timer_->defer(std::chrono::seconds(1));
   }
 
-  void start_workload()
+  void start_workload(std::function<void(void)> workload_cb)
+  {
+    auto workload_wrapper = [this, workload_cb]() {
+      auto index = workload_index_++;
+      LOG::info("workload {} starting", index);
+      workload_cb();
+      LOG::info("workload {} complete", index);
+    };
+
+    workload_threads_.emplace_back(workload_wrapper);
+  }
+
+  void start_workloads()
   {
-    thr1_ = std::thread([]() {
+    start_workload([]() {
+      system(
+          "exec 1> /tmp/workload-processes.log 2>&1; echo starting workload; set -x; whoami; pwd; ls; cd /tmp; pwd; ls; cd /; pwd; ls; cd ~; pwd; ls; echo workload complete");
+    });
+
+    start_workload([]() {
       system(
-          "echo starting workload 1; exec 1> /tmp/workload1.log 2>&1; set -x; whoami; pwd; ls; cd /tmp; pwd; ls; cd /; pwd; ls; cd ~; pwd; ls; echo workload 1 complete");
+          "exec 1> /tmp/workload-sockets.log 2>&1; echo starting workload; /root/src/test/workload/sockets/sockets.py 10 20; echo workload complete");
     });
-    thr2_ = std::thread([]() {
+
+    start_workload([]() {
       system(
-          "echo starting workload 2; /root/src/test/workload/sockets/sockets.py 10 20 > /dev/null; echo workload 2 complete");
+          "exec 1> /tmp/workload-curl.log 2>&1; echo starting workload; for n in $(seq 1 10); do curl google.com; done; echo workload complete");
```

Comment:
Does this mean this test will only pass if outbound networking and DNS is enabled for the pod running the test?

Reply:
Yes, but this test isn't really testing anything specific at the moment, just that a minimum number of messages of certain types were sent, and is meant more as a starting point for actual tests, a set of examples of things that can be tested, and a good way to investigate/learn how the kernel-collector works.